sources/Messages/Storages/Database/Sql.php

```diff
@@ -128,26 +130,84 @@ public function getAll(): Collection
         return $collection;
     }
 
-    private function getFields(): string
+    public function insert(Message $message): self
     {
-        return '
-            `message`.`id`,
-            `message`.`name`,
-            `message`.`address_place`,
-            `message`.`address_number`,
-            `message`.`address_neighborhood`,
-            `message`.`address_complement`,
-            `message`.`address_cep`,
-            `message`.`address_city_id`,
-            `message`.`address_city_name`,
-            `message`.`address_state_name`,
-            `message`.`phone`,
-            `message`.`email`,
-            `message`.`subject`,
-            `message`.`body`,
-            `message`.`date_hour`,
-            `message`.`status`
-        ';
+        $binds_array = array_keys($this->getBinds());
+        $fields = implode(", ", $binds_array);
+        $binds = implode(", ", $this->getBinds());
+
+        $statement = $this->pdo->prepare("
+            INSERT INTO {$this->table}
+            ({$fields})
+            VALUES ({$binds})
+        ");
+
+        $this->addBind($binds_array[0], PDO::PARAM_STR, $message->getName());
+        $this->addBind($binds_array[1], PDO::PARAM_STR, $message->getAddress()->getPlace());
+        $this->addBind($binds_array[2], PDO::PARAM_INT, $message->getAddress()->getNumber());
+        $this->addBind($binds_array[3], PDO::PARAM_STR, $message->getAddress()->getNeighborhood());
+        $this->addBind($binds_array[4], PDO::PARAM_STR, $message->getAddress()->getComplement());
+        $this->addBind($binds_array[5], PDO::PARAM_STR, $message->getAddress()->getCep());
+        $this->addBind($binds_array[6], PDO::PARAM_INT, $message->getAddress()->getCityId());
+        $this->addBind($binds_array[7], PDO::PARAM_STR, $message->getAddress()->getCityName());
+        $this->addBind($binds_array[8], PDO::PARAM_STR, $message->getAddress()->getStateName());
+        $this->addBind($binds_array[9], PDO::PARAM_STR, $message->getPhone());
+        $this->addBind($binds_array[10], PDO::PARAM_STR, $message->getEmail());
+        $this->addBind($binds_array[11], PDO::PARAM_STR, $message->getSubject());
+        $this->addBind($binds_array[12], PDO::PARAM_STR, $message->getBody());
+        $this->addBind($binds_array[13], PDO::PARAM_STR, $message->getDateHour()->format("Y-m-d H:i:s"));
+        $this->addBind($binds_array[14], PDO::PARAM_INT, $message->getStatus()->getValue());
+
+        $this->bind($statement);
+        
+        if ($statement->execute() === false) {
+            throw new Exception('ciebit.contactus.messages.storages.database.insert_error', 2);
+        }
+
+        return $this;
+    }
+
+    private function getBinds(): array
```

Comment:
Removeria esse método ou a forma como ele gera seu resultado, pois dessa forma existe uma redundância que nos forçará a atualizar os dados aqui e no método `getFields`

Reply:
O problema do getFields para esse caso, é que na inserção não precisamos do campo id @johnnysaymon

Reply:
Então que a fonte dos dados seja apenas uma e que antes de retornar o *array* esse método remova o valor *id*.

Reply:
A fonte dos dados seria apenas getFields então, não é mesmo? Então, na hora da contrução da query, para se adquirir os binds teria que ser feito algo assim:

`$binds = implode(", :", $this->getFields());`

Mas você concorda comigo que o primeiro campo iria ficar sem o ':', @johnnysaymon ?
Teria que ser acrescentada mais lógica.

Reply:
O incremental seria esse:

```php
$binds = ':'. implode(", :", $this->getFields());
```